File: web/src/console/common/webhook.php
<?php

namespace hutoma;

class webHook
{
    private $aiid;
    private $intentname;
    private $endpoint;
    private $enabled;

    public function __construct()
    {

    }

    public function getAiid()
    {
        return $this->aiid;
    }

    public function setAiid($aiid)
    {
        $this->aiid = $aiid;
    }

    public function getIntentName()
    {
        return $this->intentname;
    }

    public function setIntentName($intentname)
    {
        $this->intentname = $intentname;
    }

    public function getEndpoint()
    {
        return $this->endpoint;
    }

    public function setEndpoint($endpoint)
    {
        $this->endpoint = $endpoint;
    }

    public function isEnabled()
    {
        return $this->enabled;
    }

    public function enableWebHook($enabled)
    {
        $this->enabled = $enabled;
    }


    public static function fromObject($botWebHook)
    {
        $webHook = new webHook();
        $webHook->setAiid($botWebHook['aiid']);
        $webHook->setIntentName($botWebHook['intent_name']);
        $webHook->setEndpoint($botWebHook['endpoint']);
        $webHook->enableWebHook($botWebHook['enabled']);
        return $webHook;
    }

    public function toJSON()
    {
        $json = array(
            'aiid' => $this->getAiid(),
            'intent_name' => $this->getIntentName(),
            'endpoint' => $this->getEndpoint(),
            'enabled' => $this->isEnabled()
        );
        return json_encode($json);
    }
    
    public function __destruct()
    {

    }
}

File: service/src/main/java/com/hutoma/api/containers/sub/WebHook.java
package com.hutoma.api.containers.sub;

import com.google.gson.annotations.SerializedName;

import java.util.UUID;

/**
 * WebHook structure for running external code.
 */
public class WebHook {
    @SerializedName("aiid")
    private UUID aiid;
    @SerializedName("intent_name")
    private String intentName;
    @SerializedName("enabled")
    private boolean enabled;
    @SerializedName("endpoint")
    private String endpoint;

    public WebHook(UUID aiid, String intentName, String endpoint, boolean enabled) {
        this.aiid = aiid;
        this.intentName = intentName;
        this.endpoint = endpoint;
        this.enabled = enabled;
    }

    public UUID getAiid() { return this.aiid; }
    public String getIntentName() { return this.intentName; }
    public boolean isEnabled() { return this.enabled; }
    public String getEndpoint() { return this.endpoint; }
}
